Sming/Arch/Esp32/Core/HardwarePWM.cpp

```diff
@@ -140,12 +140,14 @@ uint32_t maxDuty(ledc_timer_bit_t bits)
 
 HardwarePWM::HardwarePWM(uint8_t* pins, uint8_t no_of_pins) : channel_count(no_of_pins)
 {
-	debug_d("starting HardwarePWM init");
-	periph_module_enable(PERIPH_LEDC_MODULE);
-	if((no_of_pins == 0) || (no_of_pins > SOC_LEDC_CHANNEL_NUM)) {
+	assert(no_of_pins <= SOC_LEDC_CHANNEL_NUM);
+
+	if(no_of_pins == 0) {
 		return;
 	}
```

Comment:
May I suggest:

```
assert(no_of_pins > 0 && no_of_pins <= SOC_LEDC_CHANNEL_NUM);
no_of_pins = std::min(uint8_t(SOC_LEDC_CHANNEL_NUM), no_of_pins);
```

This ensures release builds are safe (`assert` won't be compiled).

The for-loop will catch `no_of_pins == 0` so the `if` check is redundant.

Reply:
that's a good solution. thank you.